File: ios/sources/commands/ARESLineToCommand.m
//
//  ARESLineToCommand.m
//  ares
//
//  Created by PonyCui on 2018/11/29.
//  Copyright © 2018年 XT Studio. All rights reserved.
//

#import "ARESLineToCommand.h"

@implementation ARESLineToCommand

- (instancetype)initWithX:(CGFloat)x y:(CGFloat)y
{
    self = [super init];
    if (self) {
        _x = x;
        _y = y;
        [[ARESBeginPathCommand currentPath] addLineToPoint:CGPointMake(x, y)];
    }
    return self;
}

- (void)draw:(CGContextRef)ctx {
    CGContextAddLineToPoint(ctx, self.x, self.y);
}

@end

File: android/ares/src/main/java/com/xt/ares/commands/ARESLineToCommand.kt
package com.xt.ares.commands

import android.graphics.Canvas
import com.xt.ares.ARESCommand
import com.xt.ares.ARESView

class ARESLineToCommand(val x: Double, val y: Double): ARESCommand() {

    init {
        ARESBeginPathCommand.sharedPath.lineTo(x.toFloat(), y.toFloat())
    }

    override fun draw(view: ARESView, canvas: Canvas) {
        super.draw(view, canvas)
        view.currentPath.lineTo(x.toFloat(), y.toFloat())
    }

}